packages/data-stores/src/site/reducer.ts

```diff
@@ -101,15 +109,30 @@ export const sitesDomains: Reducer< { [ key: number ]: Domain[] }, Action > = (
 	return state;
 };
 
-export const launchStatus: Reducer<
-	{ [ key: number ]: { isSiteLaunched: boolean; isSiteLaunching: boolean } },
-	Action
-> = ( state = {}, action ) => {
-	if ( action.type === 'LAUNCH_SITE_START' ) {
-		return { ...state, [ action.siteId ]: { isSiteLaunched: false, isSiteLaunching: true } };
+export const launchStatus: Reducer< { [ key: number ]: SiteLaunchState }, Action > = (
+	state = {},
+	action
+) => {
+	if ( action.type === 'LAUNCH_SITE_IDLE' ) {
+		return {
+			...state,
+			[ action.siteId ]: { status: SiteLaunchStatus.IDLE, errorCode: undefined },
+		};
 	}
-	if ( action.type === 'LAUNCH_SITE_COMPLETE' ) {
-		return { ...state, [ action.siteId ]: { isSiteLaunched: true, isSiteLaunching: false } };
+	if ( action.type === 'LAUNCH_SITE_SUCCESS' ) {
+		return {
+			...state,
+			[ action.siteId ]: { status: SiteLaunchStatus.SUCCESS, errorCode: undefined },
+		};
+	}
+	if ( action.type === 'LAUNCH_SITE_FAILURE' ) {
+		return {
+			...state,
+			[ action.siteId ]: {
+				status: SiteLaunchStatus.FAILURE,
+				errorCode: SiteLaunchError.INTERNAL,
```

Comment:
Following the comments above, this could then simply become  `errorCode: action.error` , or something similar